### DISCUSSION TITLE
Message bus subscription

### DISCUSSION THREAD

**Discussion by mraj6062** [edgexfoundry/community#154] [Q&A]
1. can we subscribe to device tags or the subscription limit is up to the device itself?
2. Can we dynamically subscribe to topics, say for example if we are onboarding an AI Model in our application then the model configurations will give the topic to be subscribed can we achieve tis dynamically?

**Comment by cloudxxx8** [comment]
I am assuming the device tags is the device resources here.
You can sbubscribe topic by defining `edgex/events/<device-profile-name>/<device-name>/<source-name>`
replace `<source-name>` to your tag name
https://docs.edgexfoundry.org/3.0/design/adr/013-Device-Service-Events-Message-Bus/#messagebus-topics

Also, your application can dynamically subscribe the topics.  It depends on your implementation.
